controllers/clusters/cadence_controller.go

```diff
@@ -458,232 +441,393 @@ func (r *CadenceReconciler) handleDeleteCluster(
-	pciCompliance := c.Spec.PCICompliance
+	switch app {
+	case models.CassandraAppKind:
+		metadata.Name = models.CassandraChildPrefix + c.Name
+	case models.KafkaAppKind:
+		metadata.Name = models.KafkaChildPrefix + c.Name
+	case models.OpenSearchAppKind:
+		metadata.Name = models.OpenSearchChildPrefix + c.Name
+	}
 
-	var twoFactorDelete []*v1beta1.TwoFactorDelete
-	if len(c.Spec.TwoFactorDelete) > 0 {
-		twoFactorDelete = []*v1beta1.TwoFactorDelete{
+	return metadata
+}
+
+func (r *CadenceReconciler) reconcileAppDCs(c *v1beta1.Cadence, availableCIDRs []string, app string) (any, error) {
+	network := r.reconcileNetwork(availableCIDRs, app)
+
+	genericDataCentreSpec := v1beta1.GenericDataCentreSpec{
+		Region:              c.Spec.DataCentres[0].Region,
+		CloudProvider:       c.Spec.DataCentres[0].CloudProvider,
+		ProviderAccountName: c.Spec.DataCentres[0].ProviderAccountName,
+		Network:             network,
+	}
+
+	switch app {
+	case models.CassandraAppKind:
+		var privateIPBroadcast bool
+		if c.Spec.PrivateNetwork {
+			privateIPBroadcast = true
+		}
+
+		genericDataCentreSpec.Name = models.CassandraChildDCName
+		cassandraDCs := []*v1beta1.CassandraDataCentre{
 			{
-				Email: c.Spec.TwoFactorDelete[0].Email,
-				Phone: c.Spec.TwoFactorDelete[0].Phone,
+				GenericDataCentreSpec: genericDataCentreSpec,
+				NodeSize: c.Spec.CalculateNodeSize(
+					c.Spec.DataCentres[0].CloudProvider,
+					c.Spec.PackagedProvisioning[0].SolutionSize,
+					models.CassandraAppKind,
+				),
+				NodesNumber:                    3,
+				ReplicationFactor:              3,
+				PrivateIPBroadcastForDiscovery: privateIPBroadcast,
 			},
 		}
+
+		return cassandraDCs, nil
+
+	case models.KafkaAppKind:
+		genericDataCentreSpec.Name = models.KafkaChildDCName
+		kafkaDCs := []*v1beta1.KafkaDataCentre{
+			{
+				GenericDataCentreSpec: genericDataCentreSpec,
+				NodeSize: c.Spec.CalculateNodeSize(
+					c.Spec.DataCentres[0].CloudProvider,
+					c.Spec.PackagedProvisioning[0].SolutionSize,
+					models.KafkaAppKind,
+				),
+				NodesNumber: 3,
+			},
+		}
+		return kafkaDCs, nil
+
+	case models.OpenSearchAppKind:
+		genericDataCentreSpec.Name = models.OpenSearchChildDCName
+		openSearchDCs := []*v1beta1.OpenSearchDataCentre{
+			{
+				GenericDataCentreSpec: genericDataCentreSpec,
+				NumberOfRacks:         3,
+			},
+		}
+		return openSearchDCs, nil
 	}
-	var cassNodeSize, network string
-	var cassNodesNumber, cassReplicationFactor int
-	var cassPrivateIPBroadcastForDiscovery, cassPasswordAndUserAuth bool
-	for _, pp := range c.Spec.PackagedProvisioning {
-		cassNodeSize = pp.BundledCassandraSpec.NodeSize
-		network = pp.BundledCassandraSpec.Network
-		cassNodesNumber = pp.BundledCassandraSpec.NodesNumber
-		cassReplicationFactor = pp.BundledCassandraSpec.ReplicationFactor
-		cassPrivateIPBroadcastForDiscovery = pp.BundledCassandraSpec.PrivateIPBroadcastForDiscovery
-		cassPasswordAndUserAuth = pp.BundledCassandraSpec.PasswordAndUserAuth
+
+	return nil, nil
+}
```

Comment:
This function doesn't seem to return any errors. Please remove `error` parameter